File: apps/backend/src/grpc/services/ping.ts
import type { PingServiceServer } from "@playbook-runner/grpc/stubs"
import { logger } from "@playbook-runner/logger"

const SERVICE_NAME = "backend"

/** Mirrors `PingServicer` in `apps/ansible/app/grpc/services/ping.py`. */
export const pingServicer: PingServiceServer = {
  ping(call, callback) {
    logger.info(
      { peer: call.getPeer(), message: call.request.message },
      "grpc Ping received"
    )
    callback(null, {
      message: `pong: ${call.request.message}`,
      from_service: SERVICE_NAME,
    })
  },
}

File: apps/ansible/app/grpc/services/ping.py
from loguru import logger

from app.grpc.stubs import PingResponse, PingServiceServicer

SERVICE_NAME = "ansible"


class PingServicer(PingServiceServicer):
    async def Ping(self, request, context):
        logger.bind(peer=context.peer(), message=request.message).info("Ping recibido")
        return PingResponse(
            message=f"pong: {request.message}", from_service=SERVICE_NAME
        )
